planet/cli/subscriptions.py

```diff
@@ -262,34 +298,10 @@ async def list_subscription_results(ctx,
                                     pretty,
                                     status,
                                     limit):
-    """Gets results of a subscription and prints the API response.
-
-    This implementation is only a placeholder. To begin, instead
-    of mocking calls to the Subscriptions API, we'll use a
-    collection of fake subscriptions (the all_subs object).
-    After we refactor we will change to mocking the API.
-
-    """
-    # Begin fake subscriptions service. Note that the Subscriptions
-    # API will report missing keys differently, but the Python API
-    # *will* raise PlanetError like this.
-
-    try:
-        # Filter by status, like the Subscriptions API does.
-        if status:
-            select_results = (res
-                              for res in _get_fake_sub_results(subscription_id)
-                              if res['status'] in status)
-        else:
-            select_results = _get_fake_sub_results(subscription_id)
-
-        filtered_results = itertools.islice(select_results, limit)
-    except KeyError:
-        raise PlanetError(f"No such subscription: {subscription_id!r}")
-
-    # End fake subscriptions service. After we refactor we will get
-    # the "sub" from a method in planet.clients.subscriptions (which
-    # doesn't exist yet).
-
-    for result in filtered_results:
-        echo_json(result, pretty)
+    """Gets results of a subscription and prints the API response."""
+    async with subscriptions_client(ctx) as client:
+        filtered_results = client.list_results(subscription_id,
+                                               status=status,
+                                               limit=limit)
+        async for result in filtered_results:
+            echo_json(result, pretty)
```

Comment:
The details of the placeholder subscriptions client used to live in the CLI command but have now been moved to a PlaceholderSubscriptionsClient in the same module.

The body of the command is now quite close to its final state.